apps/server/src/modules/room/api/test/room-pass-ownership.api.spec.ts

```diff
@@ -0,0 +1,310 @@
+					const loggedInClient = await roomSetup.loginUser('SameSchoolTeacher_roomadmin');
+
+					const response = await loggedInClient.patch(`/${room.id}/members/pass-ownership`, {
+						userId: targetUser?.id,
+					});
+
+					expect(response.status).toBe(expectedStatus);
+				}
+			);
+		});
+
+		describe('when the user is a school admin from same school', () => {
+			it.each([
+				['SameSchoolTeacher_roomowner', HttpStatus.FORBIDDEN],
+				['SameSchoolTeacher_roomadmin', HttpStatus.OK],
+				['SameSchoolStudent_roomviewer', HttpStatus.BAD_REQUEST],
+				['OtherSchoolTeacher_roomeditor', HttpStatus.FORBIDDEN],
+				['SameSchoolTeacher_none', HttpStatus.FORBIDDEN],
+			] as [string, HttpStatus][])(
+				`when passing ownership to %s should return %d`,
+				async (targetUserName, expectedStatus) => {
+					const roomSetup = await setup();
+					const { room } = roomSetup;
+
+					const { adminAccount, adminUser } = UserAndAccountTestFactory.buildAdmin({ school: roomSetup.sameSchool });
+					await em.persistAndFlush([adminAccount, adminUser]);
+					em.clear();
+					const loggedInClient = await testApiClient.login(adminAccount);
+
+					const targetUser = roomSetup.getUserByName(targetUserName);
+					const response = await loggedInClient.patch(`/${room.id}/members/pass-ownership`, {
+						userId: targetUser?.id,
+					});
+
+					expect(response.status).toBe(expectedStatus);
+				}
+			);
+
+			describe('when no roomowner exists', () => {
+				it('should gracefully continue and only upgrade role of target user', async () => {
+					const roomSetup = new RoomSetup(em, testApiClient);
+					await roomSetup.setup([
+						['SameSchoolTeacher_roomadmin', 'sameSchool', 'teacher', 'roomadmin'],
+						['SameSchoolTeacher_none', 'sameSchool', 'teacher', 'none'],
+					]);
+
+					const { adminAccount, adminUser } = UserAndAccountTestFactory.buildAdmin({ school: roomSetup.sameSchool });
+					await em.persistAndFlush([adminAccount, adminUser]);
+					em.clear();
+					const loggedInClient = await testApiClient.login(adminAccount);
+
+					const targetUser = roomSetup.getUserByName('SameSchoolTeacher_roomadmin');
+					const response = await loggedInClient.patch(`/${roomSetup.room.id}/members/pass-ownership`, {
+						userId: targetUser.id,
+					});
+
+					expect(response.status).toBe(HttpStatus.OK);
+
+					const updatedRoomMembership = await loggedInClient.get(`/${roomSetup.room.id}/members-redacted`);
+					expect(updatedRoomMembership.status).toBe(HttpStatus.OK);
+					const body = updatedRoomMembership.body as RoomMemberListResponse;
+					expect(body.data).toEqual(
+						expect.arrayContaining([
+							expect.objectContaining({ userId: targetUser.id, roomRoleName: RoleName.ROOMOWNER }),
+						])
+					);
+				});
+			});
+		});
+
+		describe('when the user is a school admin from another school', () => {
+			it.each([
+				['SameSchoolTeacher_roomowner', HttpStatus.FORBIDDEN],
+				['SameSchoolTeacher_roomadmin', HttpStatus.FORBIDDEN],
```

Comment:
why are those two cases forbidden? (the upper ones)
I thought the mentioning of Christians Bug meant that those two people can receive the ownership, no?